src/azure-cli/azure/cli/command_modules/serviceconnector/action.py

```diff
@@ -32,11 +32,22 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
                     'secret_type': 'rawValue',
                     'value': v[0]
                 }
+            elif kl == 'keyvaultsecreturi':
+                d['secret_info'] = {
+                    'secret_type': 'keyVaultSecretUri',
+                    'value': v[0]
+                }
+            elif kl == 'keyvaultsecretname':
+                d['secret_info'] = {
+                    'secret_type': 'keyVaultSecretReference',
+                    'name': v[0]
+                }
             else:
-                raise ValidationError('Unsupported Key {} is provided for parameter secret_auth_info. '
-                                      'All possible keys are: name, secret'.format(k))
+                raise ValidationError('Unsupported Key {} is provided for parameter secret_auth_info. All possible '
+                                      'keys are: name, secret, keyVaultSecretUri, keyVaultSecretName'.format(k))
         if len(d) != 2:
```

Comment:
the validation is not enough, e.g, users may pass `--secret secret=xx vault-uri=xx`

Reply:
if input is `--secret secret=xx vault-uri=xx` then `len(d) != 2`